File: 백준/Silver/2839. 설탕 배달/설탕 배달.c
#include <stdio.h>

int main(void){
    int n;
    int cnt=0;
    
    scanf("%d", &n);
    
    while(n>0){
        if(n%5==0){
            n -= 5;
            cnt++;
        }
        else if(n%3==0){
            n -= 3;
            cnt++;
        }
        else if(n>5){
            n -= 5;
            cnt++;
        }
        else{
            cnt = -1;
            break;
        }
    }
    printf("%d\n", cnt);
   
}

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.io.*;
import java.util.*;
public class Main {
    private static int N;
    private static int cnt = 0;

    public static void main(String[] args) throws IOException {
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));

        N=Integer.parseInt(br.readLine());

        while(N>0) {
            if(N%5==0){
                N-=5;
                cnt++;
            } else if(N%3==0){
                N-=3;
                cnt++;
            } else{
                N-=5;
                cnt++;
            }
        }
        if(N!=0) cnt=-1;
        
        System.out.println(cnt);
    }
}